lib/test.js

```diff
@@ -859,10 +932,13 @@ Test.prototype['throws'] = function (fn, expected, msg, extra) {
 				} else if (keys.length === 0) {
 					throw new TypeError('`throws` validation object must not be empty');
 				}
-				passed = every(keys, function (key) {
+				// TS TODO: `caught.error` and `expected` should both be `object` here
+				passed = every(keys, /** @type {(key: PropertyKey) => boolean} */ function (key) {
```

Comment:
Unfortunately, TypeScript doesn't have a way of being informed that `every`'s callbacks should keep narrowing. A type assertion would be the appropriate thing to do here instead of `// @ts-expect-error` to avoid changing runtime code.

A small runtime change could be to store `error` and `expected` as variables after narrowing them.

Alternately, the next version of TypeScript might be nicer about preserving narrowings - I haven't tried.

Reply:
i'm using `next`, so it doesn't help me here. this isn't TS-expect-error tho; it's just a TODO comment

Reply:
I meant the `// @ts-expect-error`s later.